--- a/src/dawn/dawn.json	2026-08-07 15:55:40.000000000 -0700
+++ b/src/dawn/dawn.json	2026-09-10 11:36:32.693457262 -0700
@@ -4009,6 +4009,15 @@
             {"name": "window", "type": "void", "annotation": "*"}
         ]
     },
+    "dawn surface source EGL native window": {
+        "category": "structure",
+        "chained": "in",
+        "chain roots": ["surface descriptor"],
+        "tags": ["dawn"],
+        "members": [
+            {"name": "window", "type": "void", "annotation": "*"}
+        ]
+    },
     "surface source xlib window": {
         "category": "structure",
         "chained": "in",
@@ -4111,6 +4120,7 @@
             {"value": 15, "name": "compatibility mode limits"},
             {"value": 16, "name": "texture binding view dimension"},
             {"value": 17, "name": "surface source switch NWindow"},
+            {"value": 1000, "name": "dawn surface source EGL native window", "tags": ["dawn"]},
 
             {"value": 0, "name": "emscripten surface source canvas HTML selector", "tags": ["emscripten"]},
 
--- a/src/dawn/native/opengl/SwapChainEGL.cpp	2026-08-07 15:55:40.000000000 -0700
+++ b/src/dawn/native/opengl/SwapChainEGL.cpp	2026-09-10 14:22:10.479730377 -0700
@@ -203,6 +238,11 @@
     // constraints.
     auto TryCreateSurface = [&]() -> MaybeError {
         switch (surface->GetType()) {
+            case Surface::Type::EGLNativeWindow:
+                mEGLSurface = egl.CreateWindowSurface(
+                    eglDisplay, config,
+                    reinterpret_cast<EGLNativeWindowType>(surface->GetEGLNativeWindow()), attribs.data());
+                return {};
 #if DAWN_PLATFORM_IS(ANDROID)
             case Surface::Type::AndroidWindow:
                 mEGLSurface = egl.CreateWindowSurface(
--- a/src/dawn/native/Surface.cpp	2026-08-07 15:55:40.000000000 -0700
+++ b/src/dawn/native/Surface.cpp	2026-09-10 11:36:32.693457262 -0700
@@ -61,6 +61,9 @@
     const absl::FormatConversionSpec& spec,
     absl::FormatSink* s) {
     switch (value) {
+        case Surface::Type::EGLNativeWindow:
+            s->Append("EGLNativeWindow");
+            break;
         case Surface::Type::AndroidWindow:
             s->Append("AndroidWindow");
             break;
@@ -120,8 +123,14 @@
                   Branch<SurfaceDescriptorFromWindowsUWPSwapChainPanel>,
                   Branch<SurfaceDescriptorFromWindowsWinUISwapChainPanel>,
                   Branch<SurfaceSourceSwitchNWindow>,
-                  Branch<SurfaceSourceXlibWindow>, Branch<SurfaceSourceWaylandSurface>>()));
+                  Branch<SurfaceSourceXlibWindow>, Branch<SurfaceSourceWaylandSurface>,
+                  Branch<DawnSurfaceSourceEGLNativeWindow>>()));
     switch (type) {
+        case wgpu::SType::DawnSurfaceSourceEGLNativeWindow: {
+            auto* subDesc = descriptor.Get<DawnSurfaceSourceEGLNativeWindow>();
+            DAWN_INVALID_IF(subDesc->window == nullptr, "EGL native window is null.");
+            return descriptor;
+        }
 #if DAWN_PLATFORM_IS(ANDROID)
         case wgpu::SType::SurfaceSourceAndroidNativeWindow: {
             auto* subDesc = descriptor.Get<SurfaceSourceAndroidNativeWindow>();
@@ -317,9 +326,15 @@
                 Branch<SurfaceDescriptorFromWindowsUWPSwapChainPanel>,
                 Branch<SurfaceDescriptorFromWindowsWinUISwapChainPanel>,
                 Branch<SurfaceSourceSwitchNWindow>,
-                Branch<SurfaceSourceXlibWindow>, Branch<SurfaceSourceWaylandSurface>>()
+                Branch<SurfaceSourceXlibWindow>, Branch<SurfaceSourceWaylandSurface>,
+                  Branch<DawnSurfaceSourceEGLNativeWindow>>()
             .AcquireSuccess();
     switch (type) {
+        case wgpu::SType::DawnSurfaceSourceEGLNativeWindow: {
+            mType = Type::EGLNativeWindow;
+            mEGLNativeWindow = descriptor.Get<DawnSurfaceSourceEGLNativeWindow>()->window;
+            break;
+        }
         case wgpu::SType::SurfaceSourceAndroidNativeWindow: {
             auto* subDesc = descriptor.Get<SurfaceSourceAndroidNativeWindow>();
             mType = Type::AndroidWindow;
@@ -420,6 +435,11 @@
     return mMetalLayer;
 }
 
+void* Surface::GetEGLNativeWindow() const {
+    DAWN_CHECK(mType == Type::EGLNativeWindow);
+    return mEGLNativeWindow;
+}
+
 void* Surface::GetWaylandDisplay() const {
     DAWN_CHECK(mType == Type::WaylandSurface);
     return mWaylandDisplay;
--- a/src/dawn/native/Surface.h	2026-08-07 15:55:40.000000000 -0700
+++ b/src/dawn/native/Surface.h	2026-09-10 11:36:32.693502988 -0700
@@ -77,6 +77,7 @@
     // These are valid to call on all Surfaces.
     enum class Type {
         Undefined,
+        EGLNativeWindow,
         AndroidWindow,
         MetalLayer,
         WaylandSurface,
@@ -96,6 +97,7 @@
 
     // Valid to call if the type is Android
     void* GetAndroidNativeWindow() const;
+    void* GetEGLNativeWindow() const;
 
     // Valid to call if the type is WaylandSurface
     void* GetWaylandDisplay() const;
@@ -161,6 +163,8 @@
     // MetalLayer
     raw_ptr<void> mMetalLayer = nullptr;
 
+    raw_ptr<void> mEGLNativeWindow = nullptr;
+
     // ANativeWindow
     raw_ptr<void> mAndroidNativeWindow = nullptr;
 
